src/d/actor/d_a_tsubo.cpp

```diff
@@ -4,622 +4,3686 @@
+    fopAcM_SetGravity(this, sp10);
+    cXyz sp20(mStts.GetCCMoveP()->x, 0.0f, mStts.GetCCMoveP()->z);
+    if (sp20.getSquareMag() > 0.010000001f) {
+        speedF *= 0.9f;
+    }
+
+    cM3dGPla* pNorm = dComIfG_Bgsp()->GetTriPla(mAcch.m_gnd);
+    m6F0 += m814;
+    f32 fVar2;
+    f32 noGradeCos;
+    cM3dGPla* pNorm2;
+    if (pNorm != NULL) {
+        pNorm2 = pNorm;
+        fVar2 = data().m5C;
+        noGradeCos = cM_scos(cM_deg2s(data().m56 - 0.5f));
+    } else {
+        pNorm2 = NULL;
+        fVar2 = 0.0f;
+        noGradeCos = fVar2;
+    }
+
+    daObj::posMoveF_grade(this, mStts.GetCCMoveP(), &m7F4, sp0C, sp08, pNorm2->GetNP(), fVar2, noGradeCos, &m6F0);
+
+    if (mAcch.ChkGroundHit()) {
+        cLib_onBit<u32>(attention_info.flags, fopAc_Attn_ACTION_CARRY_e);
+        if (speed.abs2XZ() < 0.1f) {
+            mode_wait_init();
+        }
+    } else {
+        cLib_offBit<u32>(attention_info.flags, fopAc_Attn_ACTION_CARRY_e);
+    }
 }
 
 /* 00002D74-00002EF8       .text mode_carry_init__Q27daTsubo5Act_cFv */
-void daTsubo::Act_c::mode_carry_init() {
-    /* Nonmatching */
+void Act_c::mode_carry_init() {
+    mCyl.OffAtSPrmBit(cCcD_AtSPrm_Set_e);
+    mCyl.OnTgSPrmBit(cCcD_TgSPrm_Set_e);
+    mCyl.OffCoSPrmBit(cCcD_CoSPrm_Set_e);
+
+    mAcch.ClrRoofNone();
+    mAcch.ClrWallNone();
+    mAcch.ClrGrndNone();
+    mAcch.ClrWaterNone();
+    mAcch.OffLineCheck();
+
+    m67F = 0;
+    cLib_offBit<u32>(attention_info.flags, fopAc_Attn_ACTION_CARRY_e);
+    fopAcM_SetSpeedF(this, 0.0f);
+    spec_mode_carry_init();
+    if (mType == 2) {
+        fopAcM_seStart(this, JA_SE_OBJ_WPOT_LIFTUP, 0);
+    }
+
+    m802 = 6;
+    if (fopAcM_checkHookCarryNow(this)) {
+        m684 = 1;
+    } else {
+        m684 = 0;
+    }
+
+    if (mType == 7 && prm_get_stick()) {
+        m800 = M_attrSpine.m30;
+        se_pickup_carry_init();
+    }
+
+    cLib_offBit<u32>(base.mParameters, 0x80000000);
+    m685 = 0;
+    m686 = 0;
+    m678 = 4;
 }
 
 /* 00002EF8-0000310C       .text mode_carry__Q27daTsubo5Act_cFv */
-void daTsubo::Act_c::mode_carry() {
-    /* Nonmatching */
+void Act_c::mode_carry() {
+    daPy_py_c* player = (daPy_py_c*)daPy_getPlayerActorClass();
+    u8 tmp = m685;
+
+    if (tmp == 0) {
+        if (prm_get_spec() == 6) {
+            m685 = 1;
+            m686 = 1;
+        } else if (player->getGrabActorID() == fopAcM_GetID(this)) {
+            if (player->getGrabUpStart()) {
+                m685 = 1;
+                m686 = 1;
+            }
+        } else {
+            m685 = 1;
+            m686 = 0;
+        }
+    }
+
+    if (m685 && !tmp) {
+        init_rot_clean();
+        shape_angle.x = 0;
+        shape_angle.z = 0;
+    }
+
+#if VERSION == VERSION_DEMO
+    bool bVar2 = true;
+    if (!player->getGrabUpEnd() && player->getRightFootOnGround() == 0) {
+        bVar2 = false;
+    }
+#else
+    bool bVar2 = player->getGrabUpEnd() || player->getRightFootOnGround();
+#endif
```

Comment:
Same as above, avoid fakematching just for demo
```suggestion
    bool bVar2 = player->getGrabUpEnd() || player->getRightFootOnGround();
```